File: finstack-py/tests/golden/runners/calibration_inflation_curves.py
"""Domain runner for inflation curve calibration golden fixtures."""

from __future__ import annotations

from tests.golden.runners._placeholders import reject_flattened_outputs
from tests.golden.schema import GoldenFixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Require executable inflation curve calibration inputs before running."""
    return reject_flattened_outputs("inflation curve calibration runner", fixture)

File: finstack/valuations/tests/golden/runners/calibration_inflation_curves.rs
//! Domain runner for inflation curve calibration golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Inflation curve calibration golden runner.
pub struct CalibrationInflationCurvesRunner;

impl DomainRunner for CalibrationInflationCurvesRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::calibration_common::run_curve_fixture(fixture)
    }
}
